packages/apps/job-launcher/server/src/modules/cron-job/cron-job.service.ts

```diff
@@ -273,4 +276,50 @@ export class CronJobService {
     this.logger.log('Pending webhooks STOP');
     await this.completeCronJob(cronJob);
   }
+
+  @Cron('*/1 * * * *')
+  /**
+   * Process a pending webhook job.
+   * @returns {Promise<void>} - Returns a promise that resolves when the operation is complete.
+   */
+  public async updateJobs(): Promise<void> {
+    const lastCronJob = await this.cronJobRepository.findOneByType(
+      CronJobType.UpdateJobs,
+    );
+
+    if (lastCronJob && !lastCronJob.completedAt) {
+      return;
+    }
+    this.logger.log('Update jobs START');
+    const cronJob = await this.startCronJob(CronJobType.UpdateJobs);
+
+    try {
+      const events = await EscrowUtils.getStatusEvents(
+        this.web3Service.getValidChains(),
+        [EscrowStatus.Partial, EscrowStatus.Complete],
+        lastCronJob?.startedAt || undefined,
+      );
+
+      for (const event of events) {
+        const job = await this.jobRepository.findOneByChainIdAndEscrowAddress(
+          event.chainId,
+          ethers.getAddress(event.escrowAddress),
+        );
+
+        if (!job) continue;
+        if (event.status === EscrowStatus[EscrowStatus.Partial]) {
+          job.status = JobStatus.PARTIAL;
+          await this.jobRepository.updateOne(job);
+        } else if (event.status === EscrowStatus[EscrowStatus.Complete]) {
+          job.status = JobStatus.COMPLETED;
+          await this.jobRepository.updateOne(job);
```

Comment:
Can happened a case when job already has `COMPLETED` status?

Reply:
That case is only possible for fortune jobs, but in that case, the job will be updated to completed again. That is, it will not affect our database, but I can add a new check for the jobs that already have the updated status